pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -245,16 +245,23 @@ func (r *DNSProvider) getHostedZoneID(fqdn string) (string, error) {
 		return "", err
 	}
 
-	var hostedZoneID string
+	var zoneToID = make(map[string]string)
+	var hostedZones []string
 	for _, hostedZone := range resp.HostedZones {
 		// .Name has a trailing dot
-		if !*hostedZone.Config.PrivateZone && *hostedZone.Name == authZone {
-			hostedZoneID = *hostedZone.Id
-			break
+		if !*hostedZone.Config.PrivateZone{
```

Comment:
```suggestion
		if !*hostedZone.Config.PrivateZone {
```
small formating nit, have you ran `go fmt` on this?

Reply:
ack